File: src/FestivalPOS/src/api/printer.ts
import { jsonProperty } from "@utiliread/json";

export class Printer {
    @jsonProperty()
    id!: number;

    @jsonProperty()
    name!: string;

    @jsonProperty()
    terminalId!: number;

    @jsonProperty()
    restPrintPrinterName?: string;
}

File: src/FestivalPOS.Models/Printer.cs
using System.ComponentModel.DataAnnotations;

namespace FestivalPOS.Models;

public class Printer
{
    public int Id { get; set; }

    [MaxLength(100)]
    public required string Name { get; set; }
    public int TerminalId { get; set; }
    public Terminal? Terminal { get; set; }

    [MaxLength(50)]
    public string? RestPrintPrinterName { get; set; }

    [MaxLength(50)]
    public string? BluetoothMacAddress { get; set; }
}
